kpi/management/commands/assign_delete_submissions_to_existing_users_with_change_submissions.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+import sys
+
+from django.contrib.contenttypes.models import ContentType
+from django.core.management.base import BaseCommand
+from django.db.migrations.recorder import MigrationRecorder
+
+from kpi.constants import PERM_CHANGE_SUBMISSIONS, PERM_DELETE_SUBMISSIONS
+from kpi.models.asset import Asset
+from kpi.models.object_permission import ObjectPermission
+
+
+class Command(BaseCommand):
+
+    help = 'Assign `delete_submissions` permission to users ' \
+           'who already had `change_submissions` prior to migration ' \
+           '`0024_add_date_created_to_object_permission`'
+
+    def handle(self, *args, **options):
+
+        verbosity = options['verbosity']
+        self.stdout.write(str(verbosity))
+
+        migration_name = '0024_add_date_created_to_object_permission'
+        try:
+            expected_migration = MigrationRecorder.Migration.objects.get(
+                app='kpi',
+                name__startswith=migration_name)
+        except MigrationRecorder.Migration.DoesNotExist:
+            self.stdout.write(f'You must run migration `{migration_name}` first')
+            sys.exit(1)
+
+        applied_date = expected_migration.applied
+        asset_content_type_id = ContentType.objects.get_for_model(Asset).pk
+        query = ObjectPermission.objects.filter(
+            content_type_id=asset_content_type_id,
+            permission__codename=PERM_CHANGE_SUBMISSIONS,
+            date_created__lte=applied_date
```

Comment:
I'm not really sure we need this—provided I'm understanding it correctly. If `PERM_CHANGE_SUBMISSIONS` has allowed deletion for years, the lag time between the migration being applied and the completion of this management command isn't going to make much of a difference

Reply:
I wanted the cover the case where: 
- owners see the "delete submission" permission and do not give it to some co-workers on purpose
- the person  in charge of KoBoToolbox upgrades on their server forgot to run the management command at the same time of the migration but runs 

But I hear you. It does not hurt but it may be overkill indeed. Should I remove it?